sdks/java/io/clickhouse/src/main/java/org/apache/beam/sdk/io/clickhouse/ClickHouseWriter.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.clickhouse;
+
+import java.io.IOException;
+import java.util.List;
+import org.apache.beam.sdk.annotations.Experimental;
+import org.apache.beam.sdk.io.clickhouse.TableSchema.ColumnType;
+import org.apache.beam.sdk.values.Row;
+import org.joda.time.Days;
+import org.joda.time.Instant;
+import org.joda.time.ReadableInstant;
+import ru.yandex.clickhouse.util.ClickHouseRowBinaryStream;
+
+/** Writes Rows and field values using {@link ClickHouseRowBinaryStream}. */
+@Experimental(Experimental.Kind.SCHEMAS)
+public class ClickHouseWriter {
+  private static final Instant EPOCH_INSTANT = new Instant(0L);
+
+  @SuppressWarnings("unchecked")
+  public static void writeNullableValue(
+      ClickHouseRowBinaryStream stream, ColumnType columnType, Object value) throws IOException {
+
+    if (value == null) {
+      stream.writeByte((byte) 1);
```

Comment:
I suppose this is better to be implemented in clickhouse-jdbc. I think this is too much of implementation detail of ClickHouse to be anywhere rather than driver.

Reply:
@serebrserg agree, submitted a pull request